### PR TITLE
Moderated sessions UI message proposal

### PR DESCRIPTION
**Pull request by xacrimon**
This PR proposes some changes to the way broadcast messages are injected into a session.
This is only an RFD change and needs to be followed up by an implementation PR.

This PR also has some minor fixes to other parts of the RFD that were out of date or had minor mistakes.

cc @russjones @xinding33

### CODE REVIEW

**Comment by russjones** [comment]
@xacrimon Will these changes only apply to sessions that are moderated? I am seeing them for non-moderated sessions as well.

**Comment by xacrimon** [comment]
@russjones That is the current erroneous behaviour, this is a proposal for how it *should* be, if it gets accepted, I will make the needed changes.

**Review by zmb3 — APPROVED** [review]

File: rfd/0043-kubeaccess-multiparty.md [hunk 1]
```diff
@@ -144,18 +144,38 @@ If tap is made after the alert, the follwing message is shown:
...
+- `Session paused, waiting for additional participants...`: Sent when a session is paused due to lack of required participants.
+- `Session resumed.`: Sent when a session has the required participants and is resumed.
+- `Please tap your MFA key within 15 seconds.` and `Teleport >> MFA tap received.` are messages shown to moderators in sessions which are MFA-presence enabled.
+
 ##### Example
 
-This example illustrates how a group 3 users of which Alice is the initiator and Eve and Ben are two observers
-start a multiparty session. Below is a a series of events that happen that include what each user sees and what
-they do.
+This example illustrates how a group 3 users of which Alice is the initiator and Eve and Ben are two observers start a multiparty session. Below is a a series of events that happen that include what each user sees and actions taken.
```
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:171]: Are _observers_ always moderators? I suggest being more clear here. For example:

```suggestion
This example illustrates how a TTY-enabled SSH that requires a moderator works. In this example, Alice is a participant and both Ben and Eve are moderators. Below is a series of events that happen and what each user sees as a result.
```

**Review by xinding33 — CHANGES_REQUESTED** [review]
Reading this RFD, we use many different nouns, for example:
- Participant
- Observer
- Moderator
- Initiator

I suggest that we define exactly what nouns we want to use and be consistent in their usage. Personally, I think we can use "participant" and "moderator" and define them towards the top of the doc.

File: rfd/0043-kubeaccess-multiparty.md [hunk 2]
```diff
@@ -144,18 +144,38 @@ If tap is made after the alert, the follwing message is shown:
 Teleport >> MFA tap received.
 ```
 
+#### Broadcast messages
```
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:147]: @xacrimon changes as discussed:

1. Define all 4 nouns (participant, observer, moderator, and peer) at the top
2. Change “launching session” to “connecting to $hostname using $protocol” for clarity
3. If moderator leaves (either voluntarily or failed MFA check), participants should see “Session terminated: participant requirements not met”
4. Provide a --participant-req flag that, if set, prints participant requirements after `Session paused, waiting for additional participants...`

**Review by xinding33 — CHANGES_REQUESTED** [review]
@xacrimon see documented changes as discussed

File: rfd/0043-kubeaccess-multiparty.md [hunk 3]
```diff
@@ -3,65 +3,70 @@ authors: Joel Wejdenstål (jwejdenstal@goteleport.com)
...
+that can be used to define observers, their powers, and when and in what capacity they are required.
 
 ## Details
 
-### Multiparty sessions
+### Terminology
 
-SSH sessions via TSH currently have rich support for sessions with multiple users at once.
+This RFD repeatedly refers to some specific nouns of which explanations can be found below:
+- Participant: Any user in a session regardless of mode, including the session initiator.
```
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:28]: @xacrimon thoughts on adding definition for `initiator`?

File: rfd/0043-kubeaccess-multiparty.md [hunk 4]
```diff
@@ -3,65 +3,70 @@ authors: Joel Wejdenstål (jwejdenstal@goteleport.com)
...
+- Observer: A participant which only can view the session.
```
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:29]: Nit:

```suggestion
- Observer: A participant who can only view the session.
```

File: rfd/0043-kubeaccess-multiparty.md (hunk 3 above)
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:28]: Nit:

```suggestion
- Participant: Any user in the session regardless of mode, including the session initiator.
```

Comment (xinding33) [reply rfd/0043-kubeaccess-multiparty.md:28]: @xacrimon what do you mean by "mode' here?

File: rfd/0043-kubeaccess-multiparty.md [hunk 5]
```diff
@@ -3,65 +3,70 @@ authors: Joel Wejdenstål (jwejdenstal@goteleport.com)
...
+- Moderator: A participant who can view and terminate a session at any point.
```
Comment (xinding33) [inline rfd/0043-kubeaccess-multiparty.md:30]: Nit:

```suggestion
- Moderator: A participant who can view and terminate the session at any point.
```

**Review by xinding33 — APPROVED** [review]
A few nits but otherwise looks good to me!

File: rfd/0043-kubeaccess-multiparty.md (hunk 3 above)
Comment (xacrimon) [reply rfd/0043-kubeaccess-multiparty.md:28]: Done

Comment (xacrimon) [reply rfd/0043-kubeaccess-multiparty.md:28]: Added a definition for that too.

**Pull request merged**